### PR TITLE
ENH: external tool using new pydm entrypoints mechanism

### LINKED ISSUE #53: External Tool Plugin for PyDM
Enable `timechart` as an external tool on a widget in PyDM

### PR DESCRIPTION
**Pull request by klauer**
Marking as draft / WIP since upstream pydm will need some changes before this will work

Closes #53 

Upstream PR: https://github.com/slaclab/pydm/pull/844

### CODE REVIEW

**Comment by ZLLentz** [comment]
@klauer is this finishable now that the pydm side is merged?

**Comment by klauer** [comment]
Yes, let's get this going if all looks good.

File: timechart/tool.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        """
+
+        return {
+            "author": self.author,
+            "file": "",
+            "group": self.group,
+            "name": self.name,
+        }
+
+    def is_compatible_with(self, widget: QtWidgets.QWidget):
```
[the remark this answers is not in the document]
Comment (ZLLentz) [reply timechart/tool.py:115]: Seems like it's all good

**Review by ZLLentz — APPROVED** [review]
Let's get this in! Good feature!

No issues with this being tagged prior to pydm's entrypoint support- worst-case is a no-op

**Pull request merged**